### PR TITLE
Beta: async streaming

### PR DESCRIPTION
**Pull request by richardm-stripe**
## Summary
Provides an implementation for `request_stream_async` for `HTTPClientAsync`, so e.g. `await stripe.Quote.pdf_async(...)` will work.

```python
chunks = []
result = await stripe.Quote.pdf_async("qt_123")
async for chunk in result.stream:
    chunks.append(str(chunk, "utf-8"))
```

## Details

* There's a new public type [StripeStreamResponseAsync](https://github.com/stripe/stripe-python/pull/1233/files#diff-0c0a06a2ab83d966f0031559bfbd6de9efa84df9ffc5ebd085ea1db7f555383bR49), which is the async analog of StripeStreamResponse. (This is the type that will be returned to the user when calling `stripe.Quote.pdf_async`). The difference from StripeStreamResponse is that the stream is represented by `stream: AsyncIterable[bytes]` instead of `io: IOBase`. 
  * FYI: `IOBase` implements `Iterable[bytes]` -- so there's more symmetry between `StripeStreamResponse` and `StripeResponse` than appears at first glance.
* APIRequestor has been modified to use/produce it accordingly.
* The return type of `HTTPClientAsync.request_stream_async` is now `Tuple[AsyncIterable[bytes], ...]` and an implementation is provided for HTTPXHttpClient.

I was worried about a resource leak -- inside HTTPXHttpClient it calls `response = httpx.ASyncClient().send(...)` and then takes `content = response.aiter_bytes()` and only returns `content` to the caller. Does `response` get closed properly? I manually tested, and yes, after you do an `async for` through the bytes in `content` the response gets closed automatically.

### CODE REVIEW

File: stripe/_api_requestor.py [hunk 1]
```diff
@@ -838,6 +862,7 @@ def _interpret_streaming_response(
                 raise NotImplementedError(
                     "HTTP client %s does not return an IOBase object which "
                     "can be consumed when streaming a response."
+                    % self._get_http_client().name
```
Comment (richardm-stripe) [inline stripe/_api_requestor.py:865]: drive-by bugfix in this error message

File: stripe/_api_requestor.py [hunk 2]
```diff
@@ -266,7 +276,7 @@ def request_stream(
         )
         return resp
 
-    async def request_stream_async(
+    async def _request_stream_async(
```
Comment (anniel-stripe) [inline stripe/_api_requestor.py:279]: nit method naming consistency: Was there a specific reason to prefix this with an underscore? `APIRequestor` is not publically exposed now so every method is "private", and the other `request_...` methods in the class are not prefixed.

File: stripe/_http_client.py [hunk 3]
```diff
@@ -462,17 +463,45 @@ async def sleep_async(cls: Type[Self], secs: float) -> Awaitable[None]:
...
-        method,
-        url,
-        headers,
+        method: str,
+        url: str,
+        headers: Mapping[str, str],
         post_data,
-        is_streaming,
-        max_network_retries,
+        is_streaming: Literal[False],
```
Comment (anniel-stripe) [inline stripe/_http_client.py:473]: Why do we need this overload? Is this not the same as `is_streaming: bool` in the original method definition?

File: tests/test_api_requestor.py [hunk 4]
```diff
@@ -324,9 +332,9 @@ async def test_empty_methods_streaming_response_async(
             http_client_mock_streaming_async.assert_requested(
                 meth, post_data=post_data
             )
-            assert isinstance(resp, StripeStreamResponse)
+            assert isinstance(resp, StripeStreamResponseAsync)
 
-            assert resp.io.getvalue() == b"thisisdata"
+            assert b"".join([x async for x in resp.stream()]) == b"thisisdata"
```
Comment (anniel-stripe) [inline tests/test_api_requestor.py:337]: Do you think it would be helpful to add a `read()` method that does this for the user?

**Review by anniel-stripe — APPROVED** [review]
LGTM! Left a few nits / questions but nothing blocking

File: tests/test_api_requestor.py (hunk 4 above)
Comment (richardm-stripe) [reply tests/test_api_requestor.py:337]: Hmm, I'll consider in a follow-up.

File: stripe/_http_client.py (hunk 3 above)
Comment (richardm-stripe) [reply stripe/_http_client.py:473]: pyright complains if you only have a single overload. The other overload has a more narrow return type so if you want that one you have to include this one as well.

File: stripe/_api_requestor.py (hunk 2 above)
Comment (richardm-stripe) [reply stripe/_api_requestor.py:279]: No reason, just sloppiness/forgetfulness from me. Will revert.

Comment (richardm-stripe) [reply stripe/_api_requestor.py]: Fixed in [Revert unnecessary underscore](https://github.com/stripe/stripe-python/pull/1233/commits/f02f30fe6f2fd70a593e1369a201e34b451e431c)
